static/js/filter.js

```diff
@@ -174,17 +224,207 @@
+        }
+    }
+
+    filterToggleCategoryInput.addEventListener("click", toggleAliasCategoryBar, false);
+
+    const filterCategory = {
+        init: () => {
+            filterToggleCategoryButtonApply.addEventListener("click", filterCategory.apply, false);
+            filterToggleCategoryButtonReset.addEventListener("click", filterCategory.reset, false);
+            filterCategoryCheckboxes.forEach(checkbox => {
+                checkbox.addEventListener("change", filterCategory.oppositeCheck, false);
+            });
+            // TODO: Add "f" key listener to toggle category filter
+            // TODO: Add "esc" key listener to close category filter
+        }, 
+        reset: (e) => {
+            e.preventDefault();
+            filterCategoryCheckboxes.forEach(checkbox => {
+                checkbox.checked = false;
+            });
+
+            filterContainer.classList.remove("is-filtered-by-category");
+            
+            filterCategory.close();
+
+            
+            const isSearchActive = (filterContainer.classList.contains("is-filtered-by-search"));
+
+            // Reset back to current search query, rather than clearing all filters
+            if (isSearchActive && currentFilteredBySearchAliases && (currentFilteredBySearchAliases.length > 0) ) {
+                
+                currentFilteredBySearchAliases.forEach(alias => {
+                    alias.classList.remove("is-hidden");
+                });
+
+                return;
+            }
+            
+            
+            // Full reset
+            aliases.forEach(alias => {
+                alias.classList.remove("is-hidden");
+            });
+        },
+        apply: (e) => {
+            e.preventDefault();
+
+            const options = [];
+
+            filterCategoryCheckboxes.forEach(checkbox => {
+                if (!checkbox.checked) {
+                    return;
+                }
+
+                options.push(checkbox.dataset.categoryType);
+            });
+
+            const isSearchActive = (filterContainer.classList.contains("is-filtered-by-search"));
+
+            if (!isSearchActive) {
+                filterContainer.classList.add("is-filtered-by-category");
+            }
+
+            filterCategory.filter(options)
+            filterCategory.close();
+        },
+        close: () => {
+            toggleAliasCategoryBar();
+        },
+        oppositeCheck: (e) => {
+            const currentCategory = e.target;
+
+            if (!currentCategory.checked) {
+                return;
+            }
+            
+            const currentParentCategory = currentCategory.dataset.parentCategory;
+            
+            filterCategoryCheckboxes.forEach(checkbox => {
+                if ( (currentCategory !== checkbox) && (checkbox.dataset.parentCategory === currentParentCategory) && checkbox.checked) {
+                    checkbox.checked = !checkbox.checked;
+                }
+            });
```

Comment:
This an edgecase function to make the checkboxes from the UI behave like radio buttons in certain cases: 

Example: User opens filter and selects "show all active aliases" AND "show all disabled aliases". Since this would make the filter…useless, we detect this and toggle off the opposing checkbox.